pkg/backup/restore/restore_manager.go

```diff
@@ -66,6 +69,152 @@ func (rm *restoreManager) UpdateCondition(restore *v1alpha1.Restore, condition *
 	return rm.statusUpdater.Update(restore, condition, nil)
 }
 
+func (rm *restoreManager) pitrEnable(tc *v1alpha1.TidbCluster) error {
+	if tc.Status.TiKV.PiTRStatus.Active {
+		return nil
+	}
+
+	tc.Status.TiKV.PiTRStatus.Active = true
+	if tc.Spec.TiKV.Config == nil || tc.Spec.TiKV.Config.MP == nil {
+		tc.Spec.TiKV.Config = v1alpha1.NewTiKVConfig()
+	}
+
+	originalCfg := new(v1alpha1.PiTROverriddenConfig)
+	val := tc.Spec.TiKV.Config.Get(TiKVConfigGCThreshold)
+	if val != nil {
+		v := val.MustFloat()
+		originalCfg.GCRatioThreshold = &v
+	}
+	tc.Status.TiKV.PiTRStatus.TiKVConfigUpdateStrategy = tc.Spec.TiKV.ConfigUpdateStrategy
+
+	s := v1alpha1.ConfigUpdateStrategyInPlace
+	tc.Spec.TiKV.ConfigUpdateStrategy = &s
+	tc.Status.TiKV.PiTRStatus.OriginConfigMap = originalCfg
+	tc.Spec.TiKV.Config.Set(TiKVConfigGCThreshold, -1.0)
+	_, err := rm.deps.TiDBClusterControl.Update(tc)
+	return err
+}
+
+func (rm *restoreManager) pitrDisable(tc *v1alpha1.TidbCluster) error {
+	if tc.Status.TiKV.PiTRStatus.OriginConfigMap != nil {
+		klog.InfoS("restore-manager pitr disable, restore tikv config from origin configmap",
+			"tidbcluster", klog.KObj(tc), "originConfigMap", tc.Status.TiKV.PiTRStatus.OriginConfigMap)
+		threshold := tc.Status.TiKV.PiTRStatus.OriginConfigMap.GCRatioThreshold
+		if threshold != nil {
+			tc.Spec.TiKV.Config.Set(TiKVConfigGCThreshold, threshold)
+		} else {
+			tc.Spec.TiKV.Config.Del(TiKVConfigGCThreshold)
+		}
+		tc.Status.TiKV.PiTRStatus.OriginConfigMap = nil
+		_, err := rm.deps.TiDBClusterControl.Update(tc)
+		return err
```

Comment:
we should return an error even if `err == nil` since the whole operation is not finished here.